packages/e2e/test/web-extension/extension/background/walletManager.ts

```diff
@@ -28,30 +31,40 @@ export interface WalletFactoryDependencies {
  * Please check its documentation for examples.
  */
 const walletFactory: WalletFactory = {
-  create: async (props: WalletManagerActivateProps, { keyAgent, stores }: WalletFactoryDependencies) =>
+  create: async (props: WalletManagerActivateProps, { witnesser, stores }: WalletFactoryDependencies) =>
     (
       await getWallet({
         env,
-        keyAgent,
         logger,
         name: props.observableWalletName,
-        stores
+        stores,
+        witnesser
       })
     ).wallet
 };
 
 const storesFactory: StoresFactory = {
-  create: ({ walletId }) => storage.createPouchDbWalletStores(walletId, { logger })
+  create: ({ name }) => storage.createPouchDbWalletStores(name, { logger })
```

Comment:
Please confirm that this is compatible with existing store name (won't re-create the stores for existing wallets)

Reply:
It should be fine, the name of the argument/props object is not used for the db name but rather its content:

```
  const baseDbName = walletName.replace(/[^\da-z]/gi, '');
  const docsDbName = `${baseDbName}Docs`;
  ````

Reply:
The 'walletName' here is confusing as we're using xpub (hash?). This is how it looks now: 
![image](https://github.com/input-output-hk/cardano-js-sdk/assets/2457868/e374cdae-ee21-4f32-bdfe-3b0bfa3003ce)